airflow/jobs/job.py

```diff
@@ -130,10 +130,13 @@ def is_alive(self, grace_multiplier=2.1):
         :param grace_multiplier: multiplier of heartrate to require heart beat
             within
         """
+        if self.job_type == "SchedulerJob":
+            health_check_threshold: int = conf.getint("scheduler", "scheduler_health_check_threshold")
```

Comment:
We don't need to hard-code this, we can create an internal method on this base class that returns `health_check_threshold` and override that on `SchedulerJob` class

Reply:
There is no "SchedulerJob" class any more.

Reply:
That was the gist of the change in #31200 - to get rid of the Polymorphism we had and keep one "Job" which is not a "BaseJob" any more. We can (as next step and I will do it) to move the threshold calculation to SchedulerJobRunner, but this one is the easiest and safest fix to put specific exception in the "is_alive" method.

Reply:
BTW. There is also a follow-up from @jedcunningham to move the Job to models #31194  (where it actually belongs) and we could do it as part of that move - the whole decoupling actually made it possible to do such a move).

Reply:
This is possibly a change in behaviour.

Previously if you called `SchedulerJob.is_alive(2.2)` it would use the multiplier instead of the configures threshold.

This is probably okay to ignore the difference but should be called out in a news fragment?

Reply:
Yeah, I saw it - but this is actually not used in our code in any place and we have no SchedulerJob any more, so if someone woudl - actually use it, it would be broken (and in the Public interface description we have https://airflow.apache.org/docs/apache-airflow/stable/public-airflow-interface.html - we even now explicitly state taht Job (belonging to DB structure) is not a public interface, so we can change it's behaviour any time.

I even considered to remove the muitiplier altogether and made it a constant (Which we might actually do as a follow-up). 

I can add such newsfragment, but it woudl be misleading as it woudl refer to change in something tha does not exist any more :)

Reply:
But yeah - might be actually good to spell it out that you should use `airflow jobs check` now instead of SchedulerJob model as we know poeple DID use the queries (In Matthew's chart for one).  Let me add it.

Reply:
I've addded description. I will be weakly available this afternoon to follow-up, but I am perfectly fine with refining the message I explained in the newsfragment if someone would like to describe it better (feel free to add changes and merge without me doing so :) )

Reply:
@ashb - I think it makes little sense to mention the multiplier as a change - it's better to say (if you've used it in the past - don't do it at all).

Reply:
cc: @kaxil -> I also added another fixup, where I went a bit further - i.e. I implemented in the way that SchedulerJobRunner is overriding the threshold, so that it is not "hard-coded" (so what I discussed above) - I **think** it might be a bit nicer in the sense it is now not "hard-coded" in Job  and moved to the runner- Job is now not aware about Scheduler Job at all. Maybe that is a better idea?

Reply:
I move it back - I realized that we cannot do it - > the check **should** be based on the job_type not set externally by the runner - the way it is used by `job check` is that it retrives it from the DB - and runner is not involved.  So really the "is_alive" check must apply different threshold based on the type, not set by the runner.